proxy/proxy.go

```diff
@@ -1413,6 +1415,7 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx = newContext(lw, r, p)
 	ctx.startServe = time.Now()
 	ctx.tracer = p.tracing.tracer
+	ctx.initialSpan = span
```

Comment:
After p.do(ctx) call you can just do:
```
	p.tracing.setTag(ctx, HTTPStatusCodeTag, uint16(ctx.response.StatusCode))
```

Reply:
@szuecs at that point the `ctx.response.StatusCode` can be `nil` throwing a panic.